csrc/tensor_metadata.cpp

```diff
@@ -223,10 +226,12 @@ void validateAllocationSizesAndStrides(
     c10::IntArrayRef sizes,
     c10::IntArrayRef strides) {
   NVF_ERROR(alloc_dom.size() == contiguity.size());
+  const int64_t sizes_ndims = std::ssize(sizes);
+  const int64_t strides_ndims = std::ssize(strides);
   checkAllEqual(
-      {TensorDomain::noReductions(alloc_dom).size(),
-       sizes.size(),
-       strides.size()});
+      {std::ranges::distance(alloc_dom | TensorDomain::kNoReductions),
+       sizes_ndims,
```

Comment:
`sizes_ndims` isn't used elsewhere in the function.
```suggestion
       std::ssize(sizes),
```